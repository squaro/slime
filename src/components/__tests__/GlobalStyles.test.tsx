import { render } from '@testing-library/react';
import GlobalStyles from '../GlobalStyles';

/**
 * TODO: Apparently, assertion below doesn't work anymore as document.head returns empty <head /> tag
 * 
 * Relevant issues:
 * - https://github.com/styled-components/jest-styled-components/issues/232
 * - https://github.com/styled-components/jest-styled-components/issues/324
 */
describe.skip('GlobalStyles', (): void => {
  it('should load global styles', (): void => {
    // Act
    render(<GlobalStyles />);

    // Assert
    expect(document.head).toMatchSnapshot();
  });
});
