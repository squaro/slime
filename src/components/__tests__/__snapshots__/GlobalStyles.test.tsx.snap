// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyles should load global styles 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.2.1"
  >
    html,body,#root{width:100%;height:100%;margin:0;padding:0;overflow:hidden;}
    body{color:#fff;font-family:-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;}
    code{font-family:source-code-pro,Menlo,Monaco,Consolas,'Courier New',monospace;}
    .app{width:100%;height:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;background-color:#000;}
  </style>
</head>
`;
